### DISCUSSION TITLE
Loss Calculation of HiDream

### DISCUSSION THREAD

**Discussion by omrastogi** [bghira/SimpleTuner#1476] [Q&A]
Hi, I need help with loss calculation for HiDream, and I'm unable to find the code for it in the repo.

My current implementation(for a different problem) looks like: 
```python
noise_pred = transformer(...)
noise_pred *= -1  # the model is trained with inverted velocity :(
target = noise - model_input
logger.debug(f"Got to the end of prediction, noise_pred: {noise_pred}; target: {target}")
logger.debug("Calculating loss")
loss_weights = compute_loss_weighting_for_sd3(cfg.training.flow_match.loss_weighting_scheme, sigmas=sigmas)
loss = torch.mean((loss_weights.float() * (noise_pred.float() - target.float()) ** 2).reshape(target.shape[0], -1), 1)
loss = loss.mean()
```

Since the output gets noisier as training progresses, it seems that loss calculation might be faulty.
